### PR TITLE
grafana-agent: install as service

### PR DESCRIPTION
**Pull request by rlankfo**
Signed-off-by: Robbie Lankford <robert.lankford@grafana.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/grafana-agent.rb [hunk 1]
```diff
@@ -31,12 +34,54 @@ def install
...
+  def caveats
+    <<~EOS
+      The agent uses a configuration file that you must customize before running:
+        #{etc}/grafana-agent/config.yml
+    EOS
+  end
+
+  plist_options manual: "grafana-agent -config.file=#{HOMEBREW_PREFIX}/etc/grafana-agent/config.yml"
+
+  def plist
```
Comment (SMillerDev) [inline Formula/grafana-agent.rb:50]: The plist should be a `service do` block, and then you don't need the plist_options.

Comment (rlankfo) [reply Formula/grafana-agent.rb:50]: Ahh excellent, thanks for the info! This looks like a much cleaner way to do this.

**Comment by rlankfo** [comment]
@SMillerDev The test job is failing for me now, but it still passes locally. Is it possible to re-run this test?

File: Formula/grafana-agent.rb [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# typed: false
+# frozen_string_literal: true
+
```
Comment (SMillerDev) [inline Formula/grafana-agent.rb:3]:
```suggestion
```

**Review by SMillerDev — APPROVED** [review]
Thanks @rlankfo  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**